src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/PlayerDeathSystem.java

```diff
@@ -77,7 +81,8 @@ public void beforeDestroy(BeforeDestroyEvent event, EntityRef player, CharacterC
             updateStatistics(player, "deaths");
             dropItemsFromInventory(player);
             player.send(new RestoreFullHealthEvent(player));
-            player.send(new CharacterTeleportEvent(LASUtils.getTeleportDestination(team)));
+            Vector3f randomVector = new Vector3f(-2 + random.nextFloat()*4, -2 + random.nextFloat()*4,0);
```

Comment:
Instead of doing `random.nextFloat()*4` which generates a random number `x ∈ [0..4]` you can do `random.nextInt(5)` which generates a random number `n ∈ [0..5)` (exclusive upper bound).

Reply:
Oh, and Terasology uses the Y-axis for up/down, but I think you want to randomize the position in the x/z plane. Thus, the `0` should be the second argument.

```java
            Vector3f randomVector = new Vector3f(-2 + random.nextFloat()*4, 0, -2 + random.nextFloat()*4);
```